components/common/Layout/Layout.tsx

```diff
@@ -354,10 +374,84 @@ export const Layout = ({ children }: Props) => {
         </div>
         <div className="lg:pl-72 h-full" id="workspace">
           <main className="bg-gray-900/95 h-[100%]" id="workspace">
+            {isModalOpen && (
+              <div className="absolute z-[100] top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2">
+                <div
+                  id="popup-modal"
+                  tabIndex={-1}
+                  className="overflow-y-auto overflow-x-hidden top-0 right-0 left-0 z-50 justify-center items-center w-full md:inset-0 h-[calc(100%-1rem)] max-h-full"
+                >
+                  <div className="relative p-4 w-full max-w-md max-h-full">
+                    <div className="relative bg-white rounded-lg shadow dark:bg-gray-700">
+                      <button
+                        onClick={() => setIsModalOpen(false)}
+                        type="button"
+                        className="absolute top-3 end-2.5 text-gray-400 bg-transparent hover:bg-gray-200 hover:text-gray-900 rounded-lg text-sm w-8 h-8 ms-auto inline-flex justify-center items-center dark:hover:bg-gray-600 dark:hover:text-white"
+                        data-modal-hide="popup-modal"
+                      >
+                        <svg
+                          className="w-3 h-3"
+                          xmlns="http://www.w3.org/2000/svg"
+                          fill="none"
+                          viewBox="0 0 14 14"
+                        >
+                          <path
+                            stroke="currentColor"
+                            strokeLinecap="round"
+                            strokeLinejoin="round"
+                            strokeWidth="2"
+                            d="m1 1 6 6m0 0 6 6M7 7l6-6M7 7l-6 6"
+                          />
+                        </svg>
+                        <span className="sr-only">Close modal</span>
+                      </button>
+                      <div className="p-4 md:p-5 text-center">
+                        <svg
+                          className="mx-auto mb-4 text-gray-400 w-12 h-12 dark:text-gray-200"
+                          xmlns="http://www.w3.org/2000/svg"
+                          fill="none"
+                          viewBox="0 0 20 20"
+                        >
+                          <path
+                            stroke="currentColor"
+                            strokeLinecap="round"
+                            strokeLinejoin="round"
+                            strokeWidth="2"
+                            d="M10 11V6m0 8h.01M19 10a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
+                          />
+                        </svg>
+                        <h3 className="mb-5 text-lg font-normal text-gray-500 dark:text-gray-400">
+                          Are you sure you want to register to {peerToConnect}?
+                        </h3>
+                        <button
+                          onClick={() => {
+                            handleP2pRegister();
```

Comment:
```suggestion
                            handleP2PRegister();
```

Let's be consistent with naming